includes/class-press-sync.php

```diff
@@ -1376,4 +1383,62 @@ public function maybe_remove_post_id( $object_args ) {
 
 		return $object_args;
 	}
+
+	/**
+	 * Counts the number of taxonomies to sync.
+	 *
+	 * @since NEXT
+	 * @return int
+	 */
+	public function count_taxonomy_term_to_sync() {
+		$sql = <<<SQL
+SELECT DISTINCT
+	taxonomy, term_id
+FROM
+	{$GLOBALS['wpdb']->term_taxonomy}
+SQL;
+
+		$res = $GLOBALS['wpdb']->get_results( $sql );
+		return count( $res );
+	}
+
+	/**
+	 * Gets the next set of taxonomies/terms to sync.
+	 *
+	 * @since NEXT
+	 * @param  int   $next_page The page of results to get.
+	 * @return array
+	 */
+	public function get_taxonomy_term_to_sync( $next_page ) {
+		$offset = ( $next_page * 5 ) - 5;
+
+		$sql = <<<SQL
+SELECT DISTINCT
```

Comment:
We are not using wordpress default functions to get terms and stuff?

Reply:
WordPress built-ins don't give us good access to the actual counts of terms, and stringing everything together with the WP methods is a bit cumbersome. The direct SQL approach also  ensures that we get anything that WP might intentionally ignore for internal reasons - I don't have a specific case related to terms, but we did see something similar for users where WP was giving us bad counts based on it's internal methods because the users didn't all pass some internal checks.

Reply:
Looks like inserts are happening via wordpress functions but the reads are not. If no such function was available, I am ok with this.